pkg/server/commands/batch_check_command.go

```diff
@@ -71,14 +72,24 @@ func WithBatchCheckMaxChecksPerBatch(maxChecks uint32) BatchCheckQueryOption {
 
 func NewBatchCheckCommand(datastore storage.RelationshipTupleReader, checkResolver graph.CheckResolver, typesys *typesystem.TypeSystem, opts ...BatchCheckQueryOption) *BatchCheckQuery {
 	cmd := &BatchCheckQuery{
-		logger:              logger.NewNoopLogger(),
-		datastore:           datastore,
+		cacheController:     cachecontroller.NewNoopCacheController(),
 		checkResolver:       checkResolver,
-		typesys:             typesys,
+		datastore:           datastore,
+		logger:              logger.NewNoopLogger(),
 		maxChecksAllowed:    config.DefaultMaxChecksPerBatchCheck,
 		maxConcurrentChecks: config.DefaultMaxConcurrentChecksPerBatchCheck,
+		typesys:             typesys,
 	}
 
+	// This check query will be run against every check in the batch
+	cmd.checkCommand = NewCheckCommand(
+		cmd.datastore,
+		cmd.checkResolver,
+		cmd.typesys,
+		WithCheckCommandLogger(cmd.logger),
+		WithCacheController(cmd.cacheController),
+	)
+
```

Comment:
@jpadilla I tested out moving the `NewCheckCommand` bit inside of `NewBatchCheckCommand` and the interesting parts are the non-pointer attributes like `cacheController`.

Our pattern seems to be 1) initialize a noop verion of something 2) optionally override that in a `WithSomeOption()` command. The issue here is that the CheckCommand takes a value in its `cacheController` instead of a pointer. So if we build the CheckCommand at the  start of `NewBatchCheck`, we'll be initializing the CheckCommand with a noopCacheController(). Lmk if you want to chat irl for a few minutes to talk through this but I _think_ this is correct